src/_pytest/_code/code.py

```diff
@@ -808,7 +845,10 @@ def _truncate_recursive_traceback(self, traceback):
                 max_frames=max_frames,
                 total=len(traceback),
             )  # type: Optional[str]
-            traceback = traceback[:max_frames] + traceback[-max_frames:]
+            # Type ignored because mypy thinks the slice is a List[TracebackEntry]
+            # instead of the Traceback subclass, even though we override __getitem__().
+            # Should figure out why it does that.
+            traceback = traceback[:max_frames] + traceback[-max_frames:]  # type: ignore
```

Comment:
The slice is a `Traceback`, but only the expression `traceback[:max_frames] + traceback[-max_frames:]` is a `List[…]` then.
No idea how to fix it, but it might help you.. :)

Reply:
Right, I misunderstood the situation. Looks like a deficiency in mypy/typeshed for now. I'll update the comment.